Add unit tests for Occluder shapes

qwertychomp opened the pull request:
As requested by #43440, This PR adds unit tests for all the Occluder shapes (SphereOccluder3d, ArrayOccluder3d, PolygonOccluder3d, BoxOccluder3d, and QuadOccluder3d)

HolonProduction committed:
Autocompletion: Fix type resolution when assigning variant

HolonProduction committed:
GDScript: Fix autocompletion issues with nested types

qwertychomp commented:
My bad, i'm running clang-format over my repo to fix the code style now.

qwertychomp commented:
Squashed my commits as the docs said, please run tests again

qwertychomp commented:
Checks failed _only on linux compilation step_ due to some stuff not related to my code, that also exists in the main repo? not sure how to fix.

AThousandShips commented:
It fails because your test code is included in the wrong place, you need to include it in the `_3D_DISABLED` part with the rest of the 3D specific tests

Calinou commented on tests/scene/test_occluder_3d.h:
```diff
@@ -0,0 +1,252 @@
+
+	memdelete(test_sphere);
+}
+TEST_CASE("[SceneTree][Occluder3D] PolygonOccluder3d") {
```
```suggestion
TEST_CASE("[SceneTree][Occluder3D] PolygonOccluder3D") {
```

Calinou commented on tests/scene/test_occluder_3d.h:
```diff
@@ -0,0 +1,252 @@
+
+	memdelete(TestQuad);
+}
+TEST_CASE("[SceneTree][Occluder3D] SphereOccluder3d") {
```
```suggestion
TEST_CASE("[SceneTree][Occluder3D] SphereOccluder3D") {
```

Calinou commented on tests/scene/test_occluder_3d.h:
```diff
@@ -0,0 +1,252 @@
+}
+namespace Occluder3DTest {
+
+TEST_CASE("[SceneTree][Occluder3D] BoxOccluder3d") {
```
```suggestion
TEST_CASE("[SceneTree][Occluder3D] BoxOccluder3D") {
```

Calinou commented on tests/scene/test_occluder_3d.h:
```diff
@@ -0,0 +1,252 @@
+#include "scene/3d/occluder_instance_3d.h"
+
+#include "tests/test_macros.h"
+/*  At least on my system (windows 11 laptop, compiling using msvc)
+	when comparing two packedvector3array's, -0.0!=0.0
+	and the sphere occluder vertices for some reason have random signed zeroes sprinkled about (bug?)
+	so i have to use this function to compare the arrays using is_equal_approx  */
+bool is_packedVector3Array_equal(PackedVector3Array check_one, PackedVector3Array check_two) {
```
This is strange; it should be reported as a bug.

qwertychomp replied:
Yeah i'm going to test it later, probably just an annoying floating point inaccuracy thing.

qwertychomp commented:
Added all the capitalizations to get test case names to match node names, should finally be ready to merge now.

preslavnpetrov committed:
Move the script reload function to outside the
lock to prevent worker threads from the
ResourceLoader::load method from causing a
deadlock

atlasapplications committed:
Update Dotnet iOS Export Process

Update iOS export to allow for hybrid globalization introduced in .NET 9 while preserving .NET 8 settings. Also allows for custom ICU or no ICU.

aaronp64 committed:
Fix CPUParticles2D repeatedly scaling particles with 0 velocity and Align Y

CPUParticles2D with Align Y turned on would update particles' transforms based on the velocity when velocity was non-zero, but would not set the transform for zero velocity.  This left the transfrom value from the previous call, which may have already been scaled.  In this case, each call to _particles_process would apply the scale again, causing the particle to repeatedly grow (or shrink for scales < 1.0).

Updated the Align Y logic to always normalize the transform values before scale is applied.

zaevi committed:
C#: Fix `RefCounted` not disposed correctly in certain case.

MrBlockers committed:
Fix node cache errors on nested MultiplayerSpawner

Changes spawned_nodes to an RBSet, which preserves
insertion order after entries are removed.

This prevents simplify path commands from being
sent to clients out of order, so that the commands
for parents are recieved before the commands for
children.

This may not be the right fix (data structure, or
design).

xsellier committed:
Fix crash when using a modulo operator between a float and an integer

HolonProduction committed:
Autocompletion: Add support for global enums

HolonProduction committed:
Highlighting: Improve enum support

ryevdokimov committed:
Do not commit gizmo handles if no changes were made

Jordyfel committed:
Improve documentation of some Resource methods

Giganzo committed:
Fix Inspector checkable button sizing

Giganzo committed:
Fix PopupMenu clickable area with shadows

bruvzg committed:
Implement sparse bundle PCK support.

kuruk-mm committed:
Fix file handle leak in ZipArchive and FileAccessZip

brennennen committed:
Add error check for reflection probe invalid atlas index.

HolonProduction committed:
LSP: Extract annotations from `EditorHelp`

dalexeev committed:
GDScript: Update `get_stack()`, `print_stack()`, and `print_debug()`

bruvzg committed:
Improve edited scene layout direction detection.

IphStich committed:
Fix for Node3D request gizmos multiple times

This can result in multiple gizmos being created for the same nodes in some circumstances

juanjp600 committed:
Fix 'Script class can only be set together with base class name' error with .NET typed collections upon rebuild

adamscott committed:
[Web] Fix Web MouseWheel scrolling

pippenpaddleopsicopolis committed:
Fix select_all_on_focus behaviour on web export with virtual keyboard

limbonaut committed:
GDScript call stack as reverse linked list with fixed coroutines

* GDScript call stack as reverse linked list with issues fixed
(originally proposed in 91006).
* Fix coroutine issues with call stack by resuming async call chain
inside `GDScriptFunction::call()`.
* This fixes corrupted line numbers for coroutines in the debugger and
backtrace (106489).

Co-authored-by: Juan Linietsky <reduzio@gmail.com>

Nintorch committed:
Add support for SDL3 joystick input driver

Made possible by EIREXE, xsellier and the SDL team.

This commit includes statically linked SDL3 for Windows, Linux and macOS.
The vendored copy of SDL3 was setup to only build the required subsystems
for gamepad/joystick support, with some patches to be able to make it as
minimal as possible and reduce the impact on binary size and code size.

Co-authored-by: Álex Román Núñez <eirexe123@gmail.com>
Co-authored-by: Xavier Sellier <xsellier@gmail.com>
Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

Jordyfel committed:
Fix itemlist text separation

AThousandShips committed:
Editor: Restructure editor code

Moving various editor files into sub folders to reduce clutter

bruvzg committed:
Cleanup accessibility names.

m4gr3d committed:
Address remaining feedback on https://github.com/godotengine/godot/pull/106709

Address https://github.com/godotengine/godot/pull/106709#discussion_r2114489487

ryevdokimov committed:
Fix freelook in 3D when multiple viewports are open

Rudolph-B committed:
Fix floating point precision errors when setting particle trail length

timothyqiu committed:
Make `PROPERTY_HINT_GROUP_ENABLE` hide properties by default

ydeltastar committed:
Fix audio name doesn't appear in exports of child classes of AudioStream

Mickeon committed:
Reword `text_editor/completion/add_type_hints`'s description

dagarsar committed:
Fix mismatch of constructors

patwork committed:
Added "See Migration Guide" button to the Project Manager

mihe committed:
Fix errors not being emitted when debugger breaks on script errors

HolonProduction committed:
Autocompletion: Don't use `next` for GET_NODE inference

KoBeWi committed:
Clear current path when deselecting files

Meorge committed:
Skip killed/invalid subtween

smix8 committed:
Fix MeshInstance3D::get_active_material() error on empty mesh or empty surfaces

FixesMeshInstance3D::get_active_material() error on empty mesh or empty surfaces.

aaronfranke committed:
GDScript: Replace `abstract` keyword with `@abstract` annotation

Co-authored-by: Danil Alexeev <dalexeev12@yandex.ru>

Chaosus committed:
Prevent ternary expression with a sampler types in shaders

Chaosus committed:
Add missing input transform to visual shader preview

bruvzg committed:
[macOS] Selectively bake specific shader variants for MoltenVK.

retrotails committed:
Fix baked VoxelGI using the wrong color space

simpkins committed:
Fix jolt_physics soft body vertex normal calculation

The code previously iterated through each face and set all vertices to
that face's normal.  This resulted in each vertex getting the normal
from just one face that it belonged to (whichever face was last in this
array).  This caused weird shading artifacts.

This fixes the code so that the vertex normal is now the average normal
of all faces that it belongs to.  This results in "smooth shading"
behavior for soft body meshes.  This is still somewhat undesirable if
the input mesh was using flat shading, but it looks less bad than the
previous behavior of picking a normal at random from one attached face.

This matches the behavior of GodotPhysicsServer3D.

Fixes #107831.

m4gr3d committed:
Fix the build logic to generate the native debug symbols

DeeJayLSP committed:
Use TightLocalVector for AudioStreamWAV/MP3

Meorge committed:
Use one-based line numbers for external text editors

KoBeWi committed:
Clarify _set/_get description

Airyzz committed:
[Web] Disregard touch events in pointer callbacks

Update library_godot_input.js

fix style

bruvzg committed:
Always decode `--scene` argument UID path.

HolonProduction committed:
Autocompletion: Don't filter overrides when the existing function is the current one

Naros committed:
Expose `ScriptEditor.clear_docs_from_script`

loganapple committed:
Initial fix for double caret inserting in NodePath

Additional cases where prepending doubles up

Moved code to helper function

Added unit tests

Lookup caret character instead of passing position

Switched to using the parsed type

Adding safety checks and various cleanup

clayjohn committed:
Fix a few improper memory accesses in the clustered forward vertex shader

draw_call.instance_index should not be used directly since it doesn't take into account auto-batching

scene_data_block.data should not be used directly in the vertex shader since it can change between frames and impact motion vector generation

IN_SHADOW_PASS can only be accessed inside functions, so it needs to be a global and not a constant

MJacred committed:
Add a note about navigation mesh baking issues when using `0` for the `agent_radius`

a-johnston committed:
cleanup closed embedded processes on macos

lawnjelly committed:
`FTI` - Change `SceneTree` global setting to static

Also fixup FTI configuration warnings so that they only output when the project is using FTI.

beicause committed:
Vulkan Mobile: Fix writing vertex color in spatial shader

AThousandShips committed:
[Docs] Various grammar and spelling fixes

dalexeev committed:
GDScript: Fix call hint appearance for complex callees

BlueCube3310 committed:
Image: Fix `is_invisible` detection for RGBAH and RGBAF

kitbdev committed:
Fix ScriptEditor inline colors float handling

dalexeev committed:
GDScript: Fix double spaces for type hints when connecting signal

QbieShay committed:
Fix particles resetting properties when emitting is toggled

kitbdev committed:
Fix custom Text Editor theme changes on reload

lodetrick committed:
Add animation keying to toggleable sections

celyk committed:
Use double emulation on modelview

bruvzg committed:
[macOS] Move keyboard layout related code to base display server.

Kaleb-Reid committed:
Always send lights to sky shader if using sun scatter

KoBeWi committed:
Fix updating project settings after move

Nikitf777 committed:
Fix removing wrong vertices in the Polygon2D editor when points overlap each other

Nikitf777 committed:
Fix points jumping when dragging starts in the Polygon2D editor

beicause committed:
Fix resizing error when generating thumbnail

MJacred committed:
Add a note about navigation mesh baking issues when using `0` for the `agent_radius`

krishnabm committed:
Specified return value in several store methods in FileAccess

Some of the store methods did not specify the nature of return bool value.
Added it after referring to FileAccess.cpp method.

Moved return statement to preceding paragraph

Addressed comments: Moved return description to first line

TCROC committed:
pass -mpopcnt when using clang / llvm and targeting x86_64 arch to support popcnt forceinline methods

aaronfranke committed:
Fix OptionButton ID value range documentation

Co-authored-by: Micky <micheledevita2@gmail.com>

KoBeWi committed:
Generate scene ID for created built-in Resources

KoBeWi committed:
Mark GDScript and shader warnings for translation

adamscott committed:
[Web] Fix Webkit leak caused by the position reporting audio worklets

Co-authored-by: PizzaLovers007 <trex@parkvue.com>

Chubercik committed:
ufbx: Update to 0.20.0

geowarin committed:
Sync controller mappings DB with SDL community repo

Synced with: mdqinc/SDL_GameControllerDB@1849728

bruvzg committed:
Fix SDL threading on macOS/Linux.

And fixup pkg-config check for SDL on Linux.

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

bruvzg committed:
[SDL] Fix missing header build error on some Linux systems.

voylin committed:
Update linux .desktop file

simpkins committed:
Fix RenderingServer::mesh_surface_get_lods()

This function was incorrectly using the surface number to index into the
LOD indices vector.  This resulted in just returning the same index over
and over again.  In theory if you had a mesh with more surfaces than one
of its LOD vectors it could read pass the end of the LOD index array.

The SoftBody3D code creates a new ArrayMesh by duplicating the input
mesh, and uses `mesh_surface_get_lods()` to duplicate the LODs.  The
broken behavior here results in SoftBody3D creating broken meshes that
render nothing due to each LOD just using a single vertex.  This commit
fixes SoftBody3D to now work correctly with meshes with LODs.

Fixes #107984.

KoBeWi committed:
Restore graph_offset property

BlueCube3310 committed:
DDS: Fix loading cubemaps

paddy-exe committed:
Add permission request for Apple embedded platforms, fix microphone input

Co-Authored-By: Miguel de Icaza <miguel@gnome.org>

Supersedes https://github.com/godotengine/godot/pull/107233
Fixes https://github.com/godotengine/godot-proposals/issues/12563
Fixes https://github.com/godotengine/godot/issues/33885

Superseding Miguel's PR to get it in during the beta stage.

Calinou committed:
Document `Gradient.sample()` clamping behavior

raulsntos committed:
[.NET] Avoid exporting to duplicate architectures

Use a `HashSet` to avoid adding duplicate architectures on exporting. It seems we were adding `x86_64` and `arm64` twice to macOS exports because they are also included in the features.

Saver310 committed:
Add Polish translation for Linux .desktop file

KoBeWi committed:
Block Tween custom_step() during step

Rindbee committed:
Fix `screen_accum` not being reset when it should be in `Input::VelocityTrack`

Resets `screen_accum` like `accum`.

adamscott committed:
Revert "Web: Avoid unnecessary gamepad polling when no gamepads are connected"

This reverts commit 3e7e09f9152288ced7d404b7f7b79e8de9ad165b.

adamscott committed:
[Web] Poll controllers only if at least one is detected

Keeps the spirit of #105601

DarioSamo committed:
Fix VVL errors by changing frag_color to FP32 and dFdx/y.

KoBeWi committed:
Don't rename nodes when tree is invisible

limbonaut committed:
Add JNI variant conversion for generic Array

Chubercik committed:
meshoptimizer: Update to 0.24

stuartcarnie committed:
Apple: Use image atomic operations on supported Apple hardware

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

arlez80 committed:
Fixed generate_triangle_mesh when using PRIMITIVE_TRIANGLE_STRIP

bs-mwoerner committed:
Fix String.GetExtension() return value.

It previously returned the string itself when it should return an empty string according to the docs/examples and the C++ counterpart String::get_extension().

Fgico committed:
Gridmap editor: Removed extra mesh instantiation when setting clipboard

mihe committed:
Fix crash when disabling `SoftBody3D` while using Jolt Physics

syntaxerror247 committed:
Document that `native file dialog` is only available on Android 10+ devices.

raulsntos committed:
[.NET] Mention MSBuild panel when building fails

JekSun97 committed:
Add example of using get_connection_list_from_node

KoBeWi committed:
Fix EditorSettings usage in TLSContext

dalexeev committed:
GDScript: Fix `GDScriptLanguage::make_function()`

timothyqiu committed:
X11: Fix memory leak when using window embedding

beicause committed:
Fix Resource doesn't update when overwritten in editor

Chaosus committed:
Fix shader function overloads with incorrect order

mihe committed:
Fix crash in Jolt Physics when switching scenes in editor

Co-authored-by: Jorrit Rouwe <jrouwe@gmail.com>

simpkins committed:
Jolt physics: wake up a soft body when its transform changes

This updates `JoltSoftBody3D::set_transform()` to wake up the
soft body after changing the transform.

Previously, if you had a soft body that was sleeping in a steady state
on a ground plane, and you then translated it upwards by 1 meter it
would just hang in the air.  Now it falls to the ground correctly.

Issue #108090 has some details and an MRP (although that issue is mostly
about other problems in SoftBody3D iteself).

syntaxerror247 committed:
Android: Fix `DisplayServer.get_display_safe_area()` issues

lawnjelly committed:
FTI - Fix `MultiMesh` init and stable behaviour

lawnjelly committed:
FTI - Add reset on setting `top_level`

dogboydog committed:
fix signature in docs of EditorImportPlugin._GetOptionVisibility

Chaosus committed:
Fix subinspector for parameter preview in visual shader editor

Kazuo256 committed:
Show detach script button when added via inspector

Fixes #107059.

The SceneTreeDock was not tracking script changes in selected nodes in any capacity as far as I could assess. To do
that, my solution essentially connects the "script_changed" signal from selected nodes to
"SceneTreeDock::_update_script_button()" whenever the selection changes. It actually queues the update to make sure it
happens only once no matter how many nodes are selected.

However, only connecting that signal would leave previously selected nodes with a signal connection that should no
longer exist. To properly disconnect previously selected nodes, we have to store the list of currently selected nodes so
we can disconnect them when the selection changes.

The commit also includes some improvements to the SceneTreeDock class:

1. Remove unnecessary initialization in SceneTreeDock

This field is already initialized in the line that declares it. As such, initializing it on the constructor as well as
is redundant.

2. Queue script button updates in scene tree dock

Since we now have the option to defer the script button update and make sure it only runs once per frame, it's always
best to use the queued version of the update from a performance perspective. I'm not entirely sure if there could be any
unexpected side effects but it is a minor self-contained UI update, so it's likely a relatively safe change.

The replacement includes the bindings since it is a requirement for the other replacements in the class to work
(UndoRedo needs their method names registered in the class DB). It should be OK to remove the old non-queued bindings
too even though they are accessible in the public API because it is a "unofficial" method starting with an underscore.

stuartcarnie committed:
Metal: Use correct environment variable to generate labels

lawnjelly committed:
Clear `SceneTreeFTI` completely on enabling / disabling.

HolonProduction committed:
LSP: Don't poll during editor setup

KoBeWi committed:
Better clarify initialization of scene tiles

Rudolph-B committed:
Move occlusion culling back to being based on euclidian distance but with some distance calculation using double precision

adamscott committed:
Move Web export threads options out of variant mk2

KoBeWi committed:
Fix smoothed camera position with limits

syntaxerror247 committed:
Android: Don't exclude display cutout in immersive mode

KoBeWi committed:
Fix error spam caused by internal_vertex_count property

mihe committed:
Fix typed collections using same reference across scene instances

Miziziziz committed:
fix image download example code

using "https://placehold.co/512" as the url will cause image.load_png_from_buffer(body) to throw ERR_FILE_CORRUPT, need to specify that it's a png file in the url.

mihe committed:
Improve performance for shapeless objects when using Jolt Physics

m4gr3d committed:
Assign the `primaryHost` prior to initializing the registered Android plugins.

Address a regression introduced in https://github.com/godotengine/godot/pull/102866.
The proper behavior for Android plugins is to use `getContext` instead of `getActivity`, but since not all plugins follow this pattern, this fix provides backward compat to avoid breaking the plugins that don't.

Nikitf777 committed:
Add .zed to .gitignore

Nikitf777 committed:
Fix selecting wrong vertices in the Polygon2D editor when adding a polygon

thygrrr committed:
TabBar current tab now stays -1 when adding first tab on deselect_enable

aaronfranke committed:
Fix crash in C# bindings generator with bad enum documentation XML

lodetrick committed:
Fix Group Enable Hover Display

AThousandShips committed:
[Web] Restrict rendering method selection

Currently only the compatibility method is supported, this ensures only
it can be selected.

dmlary committed:
NavMap3D: check if obstacles have avoidance enabled

In `NavMap3D::_update_rvo_obstacles_tree_2d()` check if the
`NavObstacle3D` has avoidance enabled before adding it to the
tree.

fixes #108259

dmlary committed:
NavMap2D: check if obstacles have avoidance enabled

In NavMap2D::_update_rvo_obstacles_tree() check if the NavObstacle2D
has avoidance enabled before adding it to the tree.

This is the 2D fix for #108259

timothyqiu committed:
Clarify that Network Mode is not mandatory for plugins

syntaxerror247 committed:
Fix Immersive mode and keyboard height issues on Android

smix8 committed:
Revert "Create more optimized CSGShape3D baked static mesh"

This reverts commit 28875459d9a61885be2d3fc7c8513aa14a179ebd.

aaronfranke committed:
GLTF: Fix nasty bug with incorrect buffer indices

beicause committed:
Fix `surface_get_arrays` returns wrong index array when using empty vertex array

KoBeWi committed:
Fix filtered out nodes not getting deselected

m4gr3d committed:
Fix the EOF detection logic on Android

The current logic was causing loading to omit the last character because the EOF flag was triggered too early.

beicause committed:
SpinBox: Fix `custom_arrow_step` by snapping it to `step`

DanielSnd committed:
Add environment variable access defines to SDL linux build config

Repiteo committed:
Thirdparty: Fix SDL arm64 compilation on Windows

Rindbee committed:
Use inotify to detect devices for better reliability on Linux

Fix the issue that some devices were not detected when hot plugged.

timothyqiu committed:
Fix typo in `TextParagraph.direction` hint string

aaronfranke committed:
Move 2D and 3D particle editors to the 2D and 3D folders

akien-mga committed:
mbedTLS: Update to version 3.6.4, fixes GCC 15 compatibility

- Security fixes: https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.4
- Fixes #107754.
- Fixes #108048.

timothyqiu committed:
Fix can't remove inspector plugins after reaching max count

Mingxi-Z committed:
Fix division by zero in clearcoat

Prevents the clearcoat model from generating NaN values at grazing angles

BadMachine committed:
- fixed wrong "location" substring matching when HTTP status is 301 or 302

allenwp committed:
Always perform color correction and debanding on nonlinear sRGB values.

Fixes #107730

Co-authored-by: LuoZhihao <luo_zhihao@outlook.com>

m4gr3d committed:
Update the maven publishing configuration following the deprecation of the OSSHR service

https://central.sonatype.org/news/20250326_ossrh_sunset/

timothyqiu committed:
Fix main editor title after changing language

retrotails committed:
Fix crash when creating voxel GI data

bruvzg committed:
[RTL] Add option to scroll follow visible characters.

Rubonnek committed:
Specify Apache license version for Grisu2

dagarsar committed:
Draw guide lines over selection and focus styleboxes

Chaosus committed:
Fix lookup symbol for super()

KoBeWi committed:
Remove PropertyTweener start warning

KoBeWi committed:
Improve error when update_doc() fails

kitbdev committed:
Fix some Text Editor theme issues and clean up

bruvzg committed:
[Code Editor] Fix "Pick Color" menu option replacing multiple color items.

kitbdev committed:
Fix ScriptEditor line number in color objects

limuy2022 committed:
Fix incorrect anchor preset when custom offset is set

bruvzg committed:
[Windows export] Move debug symbol sections on export.

syntaxerror247 committed:
Fix Android splash theme regression

stuartcarnie committed:
Metal: Remove invalid assumption for image atomic operations

Fix image atomic checks, which require minimum OS version too.

Closes #108445

precup committed:
Fix issue with array comparison reference

simpkins committed:
Remove an unnecessarily include of main/ code from core/

Code in core/ generally should not depend on code from main/

This include was added in #41100 in order to call
`Main::is_project_manager()`.  The `is_project_manager()` check was
later removed in #52742 but this include was never cleaned up.

This clean-up is related to #108429.

bruvzg committed:
[RTL] Add method to get visible content bounding box.

bruvzg committed:
[RTL] Fix padding and alignment of embedded image clicks.

h1v9 committed:
Calculate md5 of import files when cache is missing. Prevents "forced" reimport

bruvzg committed:
[Windows export] Use project version as fallback.

KoBeWi committed:
Fix lingering grid from TileMapLayer editor

KoBeWi committed:
Fix export path "leaking" between presets

AeioMuch committed:
Scroll scene tree when moving item with keys

simpkins committed:
Document some deadlocks in the physics server code

Add some comments documenting locations where
PhysicsServer3D::soft_body_set_mesh() can deadlock.

godotengine/godot-proposals#12670 has a proposal for some alternate
thread-safe soft body APIs.  In the meantime it seems worth at least
documenting some of the current pitfalls in the code.

bruvzg committed:
[macOS] Selectively bake "no image atomics" shader variants.

lyuma committed:
xr: Add missing paths to /user/vive_tracker_htcx

Add 4 missing paths for wrist and ankle to XR_HTCX_vive_tracker_interaction.
These are already available in OpenXR 1.0, rev. 3 of the extension.
https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XR_HTCX_vive_tracker_interaction

AThousandShips committed:
[Codeowners] Minor improvements after restructure

WhalesState committed:
Move remaining 3D files to 3D folder and move physics gizmos to their own physics folder.

mihe committed:
Fix performance regression when rendering collision shapes

raulsntos committed:
Fix MissingNode `{get,set}_original_scene` bindings

YeldhamDev committed:
Fix unwanted resource duplication in the theme editor

Commit:
Fix error spam when hands are not tracked

mihe committed:
Fix contacts not being reported properly when using Jolt Physics

WhalesState committed:
Fix remaining physics and navigation disabling issues.

beicause committed:
Fix crash when editing some resources and reloading scene

KoBeWi committed:
Fix Variant properties losing value upon script update

bruvzg committed:
[Windows] Add SSE4.2 support runtime check.

beicause committed:
Add some multimesh null checks to avoid crash

beicause committed:
RenderingDevice: Add `uniform_type` check to avoid crash

syntaxerror247 committed:
Android Editor: Reduce default gizmo scale multiplier

The previous default value of `3` made gizmos too large and cluttered the viewport, especially on smaller screens. A value of `2` seems perfectly usable and provides a more comfortable default.

shiena committed:
Fix camera removal detection after toggling monitoring on Linux

ryevdokimov committed:
Update `auto_unfold_foreign_scenes` description

lodetrick committed:
Fix TabBar minimum size with clip_tabs on

bruvzg committed:
[RTL] Fix text selection offset in padded cells.

bruvzg committed:
[TextEdit] Draw guidelines under the text and caret.

deralmas committed:
Wayland: Fix error condition in window_get_current_screen

Leftover from when the Wayland backend was single-window only.

deralmas committed:
Disable tooltip fitting if FEATURE_SELF_FITTING_WINDOWS is available

precup committed:
Fix sphere gizmo handle position

Repiteo committed:
SCons: Support header-only modules

Calinou committed:
Document `AnimationMixer.animation_started` not being emitted for looping animations

Calinou committed:
Document Translation plurals requiring the use of the gettext PO format

stuartcarnie committed:
Editor: Fix crash when specifying `--debug-server`

The dock is not in the tree when starting the debug server via the CLI

bruvzg committed:
[macOS] Fix mouse enter/exit event and custom cursor shape in embedded game mode.

aryan-11825114 committed:
Add minimum scale to shader list

QbieShay committed:
Sort bent normal output at the bottom to prevent visual shader breakage

bruvzg committed:
[Windows] Fix SSE4.2 detection with LTO build.

Ivorforce committed:
Use `double` consistently in `Range::get_as_ratio`.

gongpha committed:
manifold: Update to upstream commit 76208dc

Changryy committed:
Fix new autoload scripts using file name instead of user defined name

Changryy committed:
Fix error when dragging non-resource file

bruvzg committed:
[CI] Use Xcode 16 for macOS build.

Alex2782 committed:
Prompt to save modified scene missing when quitting editor with running project.

bruvzg committed:
[Linux] Fix narrowing conversion error in 32-bit builds.

clayjohn committed:
Never overwrite motion vectors in the transparent pass

Alex2782 committed:
Fix: Canceling save dialog on editor exit

simpkins committed:
fix releasing the old unique name when renaming a Node

This fixes `Node::set_name()` to release the old unique name before
performing the rename.  #76560 changed the code to update `data.name`
before calling `_release_unique_name_in_owner()`, causing to incorrectly
try releasing the new name instead of the old name.

Fixes #108683

m4gr3d committed:
Fix debugger immediate disconnect

Address https://github.com/godotengine/godot/issues/108518

bruvzg committed:
[macOS] Do not use NSApplication main loop for headless mode.

bruvzg committed:
[SpriteFramesEditor] Decompress texture before auto slicing sprite sheet.

KoBeWi committed:
Fix inconsistent column in Tree click detection

Koyper committed:
Add theme colors to touch dragger and fix touch dragger icon theme

Giganzo committed:
Fix hidden scrollbar in editor settings

KoBeWi committed:
Deprecate updown icon

jcostello committed:
Fix lightmap dynamic objects with physical lights

Commit:
Add a debug version for Godot's maven central artifact

adamscott committed:
[Web] Disable GDScript LSP

bruvzg committed:
Fix support data loading with system ICU

Commit:
Fix allow any descendant to be used as a Root Type in Scene Import

bruvzg committed:
Fix editor one-click icons not showing.

aaronfranke committed:
Fix internal JSON stringify not preserving p_full_precision

m4gr3d committed:
Misc XR editor updates:
- Bump the OpenXR Vendors plugin version to `4.1.0-stable`
- Include `quest2` in the supported devices manifest config

bruvzg committed:
harfbuzz: Update to 11.3.2

mihe committed:
Prevent infinite recursion during printing

dsnopek committed:
OpenXR: Work around bug with Meta runtime on 1.0.49

qwertychomp commented:
This is a mess, I'm going to to fix up my fork then create a new PR
